### PR TITLE
Added tags to Workflow and Task Def APIs

### PR DESCRIPTION
**Pull request by coderabhigupta**
Add 6 new SDK methods for managing workflow and task tags:
- RegisterWorkflowDefWithTags
- UpdateWorkflowDefWithTags
- GetTagsForWorkflowDef
- UpdateTaskDefWithTags
- RegisterTaskDefWithTags
- GetTagsForTaskDef

### CODE REVIEW

File: sdk/model/extended_workflow_def.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ * Orkes Conductor API Server
+ *
+ * Orkes Conductor API Server
+ *
+ * API version: v2
+ * Generated by: Swagger Codegen (https://github.com/swagger-api/swagger-codegen.git)
+ */
+package model
+
+type ExtendedWorkflowDef struct {
```
Comment (v1r3n) [inline sdk/model/extended_workflow_def.go:11]: let's find a better name?  Extended Workflow Def does not give much.

File: sdk/model/extended_task_def.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ * Orkes Conductor API Server
+ *
+ * Orkes Conductor API Server
+ *
+ * API version: v2
+ * Generated by: Swagger Codegen (https://github.com/swagger-api/swagger-codegen.git)
+ */
+package model
+
+type ExtendedTaskDef struct {
```
Comment (v1r3n) [inline sdk/model/extended_task_def.go:11]: same comment as extended workflow def.

File: sdk/model/tag_object.go [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+ * Orkes Conductor API Server
+ *
+ * Orkes Conductor API Server
+ *
+ * API version: v2
+ * Generated by: Swagger Codegen (https://github.com/swagger-api/swagger-codegen.git)
+ */
+package model
+
+type TagObject struct {
```
Comment (v1r3n) [inline sdk/model/tag_object.go:11]: can we find a better name?

Comment (coderabhigupta) [reply sdk/model/tag_object.go:11]: Again, auto generated. But maybe I can create a wrapper which uses this in the API.

File: sdk/model/extended_task_def.go (hunk 2 above)
Comment (coderabhigupta) [reply sdk/model/extended_task_def.go:11]: Maybe, I can create a wrapper for Tags and pass it along as an argument in `RegisterTaskDefWithTags` `UpdateTaskDefWithTags` and from inside the functions create ExtendedTaskDef from TaskDef and this new Tag wrapper.

File: sdk/model/tag_object.go (hunk 3 above)
Comment (coderabhigupta) [reply sdk/model/tag_object.go:11]: Added a new class called MetadataTag which needs to be passed to the new methods now.

File: sdk/model/extended_task_def.go (hunk 2 above)
Comment (coderabhigupta) [reply sdk/model/extended_task_def.go:11]: Made the change mentioned above.

File: sdk/model/extended_workflow_def.go (hunk 1 above)
Comment (coderabhigupta) [reply sdk/model/extended_workflow_def.go:11]: Started using this internally instead of asking the user to pass it.

**Review by c4lm — APPROVED** [review]

**Pull request merged**